internal/db/schema/migrations/oss/postgres/64/01_ssh_targets.up.sql

```diff
@@ -0,0 +1,363 @@
+     -- max duration of the session in seconds.
+     -- default is 8 hours
+    session_max_seconds int not null default 28800
+      constraint session_max_seconds_must_be_greater_than_0
+      check(session_max_seconds > 0),
+    -- limit on number of session connections allowed. -1 equals no limit
+    session_connection_limit int not null default 1
+      constraint session_connection_limit_must_be_greater_than_0_or_negative_1
+      check(session_connection_limit > 0 or session_connection_limit = -1),
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version,
+    worker_filter wt_bexprfilter,
+    egress_worker_filter wt_bexprfilter,
+    ingress_worker_filter wt_bexprfilter,
+    constraint target_ssh_project_id_name_uq
+      unique(project_id, name) -- name must be unique within a project scope.
+  );
+  comment on table target_ssh is
+    'target_ssh is a table where each row is a resource that represents an ssh target. '
+    'It is a target subtype.';
+
+  drop trigger if exists insert_target_subtype on target_ssh;
+  create trigger
+    insert_target_subtype
+  before insert on target_ssh
+    for each row execute procedure insert_target_subtype();
+
+  drop trigger if exists delete_target_subtype on target_ssh;
+  create trigger
+    delete_target_subtype
+  after delete on target_ssh
+    for each row execute procedure delete_target_subtype();
+
+  -- define the immutable fields for target
+  drop trigger if exists immutable_columns on target_ssh;
+  create trigger immutable_columns before update on target_ssh
+    for each row execute procedure immutable_columns('public_id', 'project_id', 'create_time');
+
+  drop trigger if exists update_version_column on target_ssh;
+  create trigger update_version_column after update on target_ssh
+    for each row execute procedure update_version_column();
+
+  drop trigger if exists update_time_column on target_ssh;
+  create trigger update_time_column before update on target_ssh
+    for each row execute procedure update_time_column();
+
+  drop trigger if exists default_create_time_column on target_ssh;
+  create trigger default_create_time_column before insert on target_ssh
+    for each row execute procedure default_create_time();
+
+  drop trigger if exists update_ssh_target_filter_validate on target_ssh;
+  create trigger update_ssh_target_filter_validate before update on target_ssh
+    for each row execute procedure validate_filter_values_on_update();
+
+  drop trigger if exists insert_ssh_target_filter_validate on target_ssh;
+  create trigger insert_ssh_target_filter_validate before insert on target_ssh
+    for each row execute procedure validate_filter_values_on_insert();
+
+  insert into oplog_ticket (name, version)
+  values ('target_ssh', 1)
+  on conflict do nothing;
+
+  -- Replaces view from 60/03_wh_sessions.up.sql
+  drop view if exists whx_host_dimension_source;
+  create view whx_host_dimension_source as
```

Comment:
I don't think you can create these views again until you have replaced the `target_all_subtypes`, otherwise the attempt to drop `target_all_subtypes` will fail.

Reply:
Yeah, I just noticed I screwed that up, sorry 😂

Reply:
I think you should, drop all 3 views. Then create `target_all_subtypes` then create these `whx` views.

Reply:
There we go: https://github.com/hashicorp/boundary/pull/2960/commits/0fda616a328cdb0c82d190366e77c25f8a63dd70